storage/innobase/trx/trx0sys.cc

```diff
@@ -1253,15 +1280,42 @@ ulint
 trx_sys_any_active_transactions(void)
 /*=================================*/
 {
-	ulint	total_trx = 0;
-
 	mutex_enter(&trx_sys->mutex);
 
-	total_trx = UT_LIST_GET_LEN(trx_sys->rw_trx_list)
-		  + UT_LIST_GET_LEN(trx_sys->mysql_trx_list);
+	ulint	total_trx = UT_LIST_GET_LEN(trx_sys->mysql_trx_list);
+
+	if (total_trx == 0) {
```

Comment:
Yes, I agree. It looks like the calling side really cares only about the return value of this function being zero/non-zero. The value itself is used for diagnostics only.